### PR TITLE
chore(engine): dedup consistent forkchoice state check

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Deduplicate consistent forkchoice state check.

Previously, there was a separate check in the payload attributes handling branch because of early return. We can forkchoice state consistency before that because this is the only path with valid payload status.

### CODE REVIEW

File: crates/consensus/beacon/src/engine/mod.rs [hunk 1]
```diff
@@ -407,15 +407,15 @@ where
                     }
                 }
 
-                if let Some(attrs) = attrs {
-                    // if we return early then we wouldn't perform these consistency checks, so we
-                    // need to do them here, and should do them before we process any payload
```
Comment (mattsse) [inline crates/consensus/beacon/src/engine/mod.rs:412]: this looks very relevant,

unsure we can change the order without violating engine tests

**Review by mattsse — COMMENTED** [review]
we should confirm that this does not violate any engine tests

File: crates/consensus/beacon/src/engine/mod.rs (hunk 1 above)
Comment (rkrasiuk) [reply crates/consensus/beacon/src/engine/mod.rs:412]: we can check for sure, but this is a pure control flow change. if we perform the check before handling the attributes we no longer return early

**Review by Rjected — APPROVED** [review]
lgtm pending no hive regressions

**Comment by rkrasiuk** [comment]
@mattsse @Rjected dispatched hive workflow
https://github.com/paradigmxyz/reth/actions/runs/8707855109

**Review by mattsse — APPROVED** [review]

**Comment by Rjected** [comment]
cool, all previously passing `Inconsistent...` and `Unknown ...` tests still pass

**Pull request merged**